dev/release-util/src/commit-summary.js

```diff
@@ -0,0 +1,136 @@
+const dotenv = require('dotenv');
+const { docopt } = require('docopt');
+const { exec } = require('child_process');
+const fs = require('fs/promises');
+
+dotenv.config(); // load from .env
+const username = process.env.JIRA_USERNAME;
+const apiKey = process.env.JIRA_API_KEY;
+const baseUrl = process.env.JIRA_BASE_URL;
+const auth = Buffer.from(`${username}:${apiKey}`).toString('base64');
+const githubRepoUrl = process.env.GITHUB_REPO_URL;
+
+const docs = `
+Usage:
+    commit-summary <from> [<to>]
+
+        <from> - the git tag to check from (e.g. last release)
+        <to> - the git tag to check upto, defaults to main
+`;
+
+/**
+ * a cli tool to aid with finding a good release candidate/tag
+ * - pulls in commits from a given tag
+ * - fetches ticket numbers from commit(scope)
+ * - prints commit + ticket status
+ */
+async function run() {
+	const argv = docopt(docs);
+	const from = argv['<from>'];
+	const to = argv['<to>'] || 'main';
+	const changes = await runCommand(`git log --pretty="%H; %s" ${from}...${to}`);
+
+	const commits = changes
+		.split('\n') // split into lines
+		.filter(Boolean); // filter out blank lines
+
+	const rows = [`commit,ticket,title,status,commit link`];
+	const tickets = new Map();
+	for await (const commit of commits) {
+		const hash = commit.substring(0, 40);
+		const message = commit.substring(42);
+		const match = message.match(/^.*\((.*)\):.*/);
```

Comment:
is it worth extracting the regexes into named variables to simplify reading
e.g. commitizenMessageFormat

Reply:
See [dev/release-util/src/commit.js](https://github.com/Planning-Inspectorate/appeal-planning-decision/pull/3301/files#diff-9063fbe57143fe363f32a2056bbfb2d9143f3528bbbc03a8aefcc642d6ff2d5a)